solr/core/src/java/org/apache/solr/update/SolrIndexSplitter.java

```diff
@@ -765,18 +766,11 @@ static FixedBitSet[] split(
     return docSets;
   }
 
-  public static String getRouteKey(String idString) {
-    int idx = idString.indexOf(CompositeIdRouter.SEPARATOR);
-    if (idx <= 0) return null;
-    String part1 = idString.substring(0, idx);
-    int commaIdx = part1.indexOf(CompositeIdRouter.bitsSeparator);
-    if (commaIdx > 0 && commaIdx + 1 < part1.length()) {
-      char ch = part1.charAt(commaIdx + 1);
-      if (ch >= '0' && ch <= '9') {
-        part1 = part1.substring(0, commaIdx);
-      }
+  private static void checkRouterSupportsSplitKey(HashBasedRouter hashRouter, String splitKey) {
```

Comment:
The expectation is much clearer with this method.
I'm not familiar with the other DocRouter. Does this mean that split is not supported at all with other DocRouter types?

Reply:
the other one is PLAIN router. I don't think Split is possible for that